presto-main/src/main/java/com/facebook/presto/dispatcher/DefaultQueryPrerequisites.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.dispatcher;
+
+import com.facebook.presto.spi.QueryId;
+import com.facebook.presto.spi.prerequisites.QueryPrerequisites;
+import com.facebook.presto.spi.prerequisites.QueryPrerequisitesContext;
+
+import java.util.concurrent.CompletableFuture;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+
+public class DefaultQueryPrerequisites
+        implements QueryPrerequisites
+{
+    private static final CompletableFuture<?> COMPLETED_FUTURE = completedFuture(null);
```

Comment:
curious why we are using CompletableFuture instead of ListenableFuture
see https://github.com/prestodb/presto/wiki/Presto-Development-Guidelines#development
```
Use ListenableFuture in preference to CompletableFuture. The former has a much simpler API and is less error prone than the latter. We previously used CompletableFuture everywhere after updating to Java 8, then switched back to ListenableFuture after encountering many bugs caused by the non-obvious design choices in the CompletableFuture and CompletionStage APIs.
```

Reply:
`presto-spi` which defines the interface and hence the return type does not take  a dependency on guava and hence every interface in the SPI uses `CompletableFuture`